app/controllers/main/routes.py

```diff
@@ -43,12 +43,26 @@ def redirectToLogout():
 @main_bp.route('/', methods=['GET'])
 def landingPage():
     managerProgramDict = getManagerProgramDict(g.current_user)
-    eventsInTerm = list(Event.select().where(Event.term == g.current_term, Event.isCanceled == False))
-    programsWithEventsList = [event.program for event in eventsInTerm if not event.isPast]
 
-    return render_template("/main/landingPage.html", managerProgramDict = managerProgramDict,
-                                                     term = g.current_term,
-                                                     programsWithEventsList = programsWithEventsList)
+    # Optimize the query to fetch programs with non-canceled, non-past events in the current term
+    programsWithEventsQuery = (Program
+        .select()
+        .join(Event)
+        .where(
+            (Event.term == g.current_term) &
+            (Event.isCanceled == False) &
+            (Event.isPast == False)
+        )
+        .distinct()  # Ensure only unique programs are included
```

Comment:
Please put the query in the same format as all other queries in the application.

Reply:
Done.